src/opal.c

```diff
@@ -759,13 +760,87 @@ binary_unary (char compound_char, lexeme_type_e compound_type,
 lexeme_s
 get_identifier_lexeme (int char_line, int char_col)
 {
-
-  // TODO: Replace stub implementation
   lexeme_s retVal =
       { 0 };
+  retVal.line=char_line;
+  retVal.column=char_col;
+  char identifier_str[1024] = {0};
+  int str_len=0;
+  bool regex_match = false;
+
+  ///Get string to analyze
+  while (isalnum(next_char) || next_char == '_')
+    {
+      identifier_str[str_len] = next_char;
+      str_len++;
+      read_next_char();
+    }
+
+  ///terminate string
+  identifier_str[str_len]='\0';
+  str_len++;
+
+  ///Determine if string is a reserved keyword
+  for(int i=0; i<5; i++)
+    {
+      if(strcmp(identifier_str,keyword_arr[i].str) == 0){
+
+          retVal.char_val=keyword_arr[i].str;
+          retVal.type=keyword_arr[i].lex_type;
+          return retVal;
+      }
+    }
+
+  ///Determine if the string is an integer via regex
+  regex_match = match(identifier_str, int_regex_pattern);
+  if(regex_match == 1)
+    {
+      int intVal = strtol(identifier_str, NULL, 0);
```

Comment:
Please set `errno` to 0 before calling strtol. `man strtol` has an excellent explanation why and also a good example.

```
errno = 0;    /* To distinguish success/failure after call */
intVal = strtol(identifier_str, NULL, 0);

/* Check for various possible errors */
if ((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN))
    || (errno != 0 && val == 0))
  {
    perror(identifier_str);
    return (errno);
  }
```
You will need to include limits.h for INT_MAX